store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -4868,6 +4868,21 @@ public void deleteDistributionList(String zimbraId) throws ServiceException {
     }
 
     private void deleteDistributionList(LdapDistributionList dl) throws ServiceException {
+        deleteDistributionList(dl, false);
+    }
+
+    private void deleteDistributionList(LdapDistributionList dl, boolean cascadeDelete) throws ServiceException {
+        // check if cascadeDelete is true. If it's true, delete all subgroups.
+        if (cascadeDelete) {
+            Set<String> members = dl.getAllMembersSet();
+            for (String member : members) {
+                LdapDistributionList subDl = (LdapDistributionList) getDistributionListByNameInternal(member);
```

Comment:
Chinmay I believe you have to handle if any of the members is just an account.

Reply:
This is taken care in existing code.